api_v3/lib/types/edgeServer/KalturaEdgeServer.php

```diff
@@ -0,0 +1,248 @@
+	/**
+	 * Id of the parent edge server
+	 * 
+	 * @var int
+	 * @filter eq,in
+	 */
+	public $parentId;
+	
+	private static $map_between_objects = array
+	(
+		"id",
+		"createdAt",
+		"updatedAt",
+		"partnerId",
+		"name",
+		"systemName",
+		"desciption",
+		"status",
+		"tags",
+		"hostName",
+		"playbackHostName",
+		"deliveryProfileIds",
+		"parentId",
+	);
+	
+	/* (non-PHPdoc)
+	 * @see KalturaObject::getMapBetweenObjects()
+	 */
+	public function getMapBetweenObjects ( )
+	{
+		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
+	}	
+	
+	/* (non-PHPdoc)
+	 * @see KalturaObject::toInsertableObject()
+	 */
+	public function toInsertableObject ( $object_to_fill = null , $props_to_skip = array() )
+	{
+		if(is_null($object_to_fill))
+			$object_to_fill = new EdgeServer();
+			
+		return parent::toInsertableObject($object_to_fill, $props_to_skip);
+	}
+	
+
+	/* (non-PHPdoc)
+	 * @see KalturaObject::validateForInsert()
+	 */
+	public function validateForInsert($propertiesToSkip = array())
+	{
+		$this->validateMandatoryAttributes();
+			
+		if($this->systemName)
+		{
+			$c = KalturaCriteria::create(EdgeServerPeer::OM_CLASS);
+			$c->add(EdgeServerPeer::SYSTEM_NAME, $this->systemName);
+			if(StorageProfilePeer::doCount($c))
+				throw new KalturaAPIException(KalturaErrors::SYSTEM_NAME_ALREADY_EXISTS, $this->systemName);
+		}
+	
+		return parent::validateForInsert($propertiesToSkip);
+	}
+
+	/* (non-PHPdoc)
+	 * @see KalturaObject::validateForUpdate()
+	 */
+	public function validateForUpdate($sourceObject, $propertiesToSkip = array())
+	{
+		$this->validateMandatoryAttributes();
+		
+		if($this->systemName)
+		{
+			$c = KalturaCriteria::create(EdgeServerPeer::OM_CLASS);
+			$c->add(EdgeServerPeer::ID, $sourceObject->getId(), Criteria::NOT_EQUAL);
+			$c->add(EdgeServerPeer::SYSTEM_NAME, $this->systemName);
+			if(EdgeServerPeer::doCount($c))
+				throw new KalturaAPIException(KalturaErrors::SYSTEM_NAME_ALREADY_EXISTS, $this->systemName);
+		}
+		
+		return parent::validateForUpdate($sourceObject, $propertiesToSkip);
+	}
+	
+	public function validateMandatoryAttributes()
+	{
+		$this->validatePropertyMinLength("name", 1, true);
+		$this->validatePropertyNotNull("hostName");
+	}
+	
+	/* (non-PHPdoc)
+	 * @see KalturaObject::toObject()
+	 */
+	public function toObject ( $object_to_fill = null , $props_to_skip = array() )
+	{
+		if(is_null($object_to_fill))
+			$object_to_fill = new EdgeServer();
+		
+		
+		$object_to_fill =  parent::toObject($object_to_fill, $props_to_skip);
+		
+		// Delivery Profile Ids
+		$deliveryProfileIds = $this->deliveryProfileIds;
+		
+		$deliveryProfiles = array();
+		if($deliveryProfileIds)
+			foreach($deliveryProfileIds->toArray() as $keyValue) 
+				$this->insertObject($deliveryProfiles, $keyValue->key, $keyValue->value);
+			
+		$object_to_fill->setDeliveryProfileIds($deliveryProfiles);
+		
+		return $object_to_fill;
+	}
+	
+	protected function insertObject(&$res, $key, $value) {
```

Comment:
You define an internal language here, I think you better use well structured API object instead.